equinix/data_source_fabric_service_profile_acc_test.go

```diff
@@ -0,0 +1,114 @@
+package equinix_test
+
+import (
+	"fmt"
+	"github.com/equinix/terraform-provider-equinix/internal/acceptance"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"testing"
+)
+
+func testAccFabricReadServiceProfileConfig(spName string, portUUID string, portType string, portMetroCode string) string {
+	return fmt.Sprintf(`
+
+resource "equinix_fabric_service_profile" "test" {
+  name = "%s"
+  description = "Generic Read SP"
+  self_profile = false
+  type = "L2_PROFILE"
+  notifications {
+      emails = ["opsuser100@equinix.com"]
+      type = "BANDWIDTH_ALERT"
+  }
+  tags = ["VoIP", "Saas"]
+  visibility = "PRIVATE"
+  ports {
+      uuid = "%s"
+      type = "%s"
+      location {
+        metro_code = "%s"
+      }
+      cross_connect_id = ""
+      seller_region = ""
+      seller_region_description = ""
+  }
+  access_point_type_configs {
+      type = "COLO"
+      connection_redundancy_required = false
+      allow_bandwidth_auto_approval = false
+      allow_remote_connections = false
+      connection_label = "test"
+      enable_auto_generate_service_key = false
+      bandwidth_alert_threshold=  10
+      allow_custom_bandwidth = true
+      api_config {
+        api_available = false
+        equinix_managed_vlan = true
+        bandwidth_from_api = false
+        integration_id = "test"
+        equinix_managed_port = true
+      }
+      authentication_key{
+        required = false
+        label = "Service Key"
+        description = "XYZ"
+      }
+      supported_bandwidths = [100,500]
+  }
+  marketing_info {
+    promotion = false
+  }
+}
+
+data "equinix_fabric_service_profile" "test" {
+		uuid = equinix_fabric_service_profile.test.uuid
+}`, spName, portUUID, portType, portMetroCode)
+}
+
+func TestAccFabricReadServiceProfileByUuid_PFCR(t *testing.T) {
+	ports := GetFabricEnvPorts(t)
+
+	var portUuid, portMetroCode string
+	if len(ports) > 0 {
+		portUuid = ports["pfcr"]["dot1q"][0].Uuid
+		portType = ports["pfcr"]["dot1q"][0].type
```

Comment:
The variable was never defined before assignment, and this will fail because `.type` is not part of the v4.Port struct. Please update.